framework/src/modules/token/commands/transfer.ts

```diff
@@ -56,20 +105,63 @@ export class TransferCommand extends BaseCommand {
 
 	public async execute(context: CommandExecuteContext<Params>): Promise<void> {
 		const { params } = context;
-		// TODO: Fix in https://github.com/LiskHQ/lisk-sdk/issues/7573
-		await this._method.initializeUserAccount(
-			context.getMethodContext(),
-			params.recipientAddress,
-			params.tokenID,
-			context.transaction.senderAddress,
-			BigInt(50000000),
-		);
-		await this._method.transfer(
-			context.getMethodContext(),
-			context.transaction.senderAddress,
-			params.recipientAddress,
-			params.tokenID,
-			params.amount,
+
+		const userStore = this.stores.get(UserStore);
+
+		const recipientAccountKey = userStore.getKey(params.recipientAddress, params.tokenID);
+
+		const recipientAccountExists = await userStore.has(context, recipientAccountKey);
+
+		if (!recipientAccountExists) {
+			await this._intializeUserStore(
+				context,
+				params.recipientAddress,
+				params.tokenID,
+				context.transaction.senderAddress,
+			);
+		}
+
+		const senderAccountKey = userStore.getKey(context.transaction.senderAddress, params.tokenID);
+		const senderAccount = await userStore.get(context, senderAccountKey);
+		const recipientAccount = await userStore.get(context, recipientAccountKey);
+
+		senderAccount.availableBalance -= params.amount;
+		recipientAccount.availableBalance += params.amount;
+
+		await userStore.save(context, context.transaction.senderAddress, params.tokenID, senderAccount);
+		await userStore.save(context, params.recipientAddress, params.tokenID, recipientAccount);
```

Comment:
1. Sender available balance should be checked against params.amount because it's possible that before this `execute` function, sender paid the fee and there is no remaining balance
2. sender should be updated and saved, then recipient should be fetched and deduct the balance because there is a case where sender and recipient are the same